### PR TITLE
LayerZero V2 Adapter

### PR DESCRIPTION
**Pull request by jboetticher**
Adds a LayerZero V2 adapter. Fixes LayerZero dependency clashing with our own forks of the project.

### CODE REVIEW

File: test/contracts/mocks/lz/LayerZeroV2Mock.sol [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+        (bool success, ) = payable(_refundAddress).call{value: msg.value}("");
+        if (!success) {
+            revert LayerZeroV2GMPMock__RefundAddressDoesNotReceiveRefund();
+        }
+
+        return MessagingReceipt(messageId, nonce - 1, MessagingFee(msg.value, 0));
+    }
+
+    function getChainId() public pure returns (uint32) {
+        return 1;
```
Comment (gusjavaz) [inline test/contracts/mocks/lz/LayerZeroV2Mock.sol]: Should return 31337 for hardhat deployment tests to work

**Comment by jboetticher** [comment]
NPM installation error occurs occasionally due to the way that the forks are represented. :/

File: test/contracts/mocks/lz/LayerZeroV2Mock.sol (hunk 1 above)
Comment (jboetticher) [reply test/contracts/mocks/lz/LayerZeroV2Mock.sol]: Which tests? This should be left alone, as LayerZero needs to set its own value chain ID. If I remove this the forge tests fail 😆

**Comment by jboetticher** [comment]
Line coverage fails because of LZV1 still being there + LZV2OApp 😢

**Pull request merged**